src/app/pages/courses/components/add-course/add-course.component.ts

```diff
@@ -0,0 +1,12 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+    selector: 'app-add-course',
+    templateUrl: './add-course.component.html',
+    styleUrls: ['./add-course.component.scss'],
```

Comment:
I suggest to use ChangeDetection OnPush for all the components

https://mokkapps.de/blog/the-last-guide-for-angular-change-detection-you-will-ever-need/#avoiding-change-detection-loops-and-expressionchangedaftercheckederror